core/src/main/java/org/mule/api/metadata/MuleMetadataManager.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.api.metadata;
+
+import org.mule.api.MuleContext;
+import org.mule.api.context.MuleContextAware;
+import org.mule.api.metadata.descriptor.OperationMetadataDescriptor;
+import org.mule.api.source.MessageSource;
+import org.mule.construct.Flow;
+import org.mule.extension.api.metadata.MetadataResolvingException;
+import org.mule.metadata.api.model.MetadataType;
+import org.mule.util.metadata.ResultFactory;
+
+import java.util.List;
+
+public class MuleMetadataManager implements MetadataManager, MuleContextAware
+{
+
+    public static final String MANAGER_REGISTRY_ID = "core.metadata.manager.1";
```

Comment:
if the user never should look this up in the registry and instead always inject, we should remove this.